### PR TITLE
issues related with multiple charts in story

### PR DESCRIPTION
**Pull request by mmilaprat**
when you add multiple charts in a story some issues appears like the one with labels in line charts, this PR solve these issues, it's related with issue 678 (policycompass/policycompass/issues/678)

### CODE REVIEW

File: app/modules/visualization/controllers/visualizationControllers.js [hunk 1]
```diff
@@ -791,75 +791,77 @@ angular.module('pcApp.visualization.controllers.visualization', [
...
-                            'initialLng': initialLng,
-                            'initialZoom': initialZoomMap,
-                            'showBubbles': $scope.showBubbles,
-                            'showMovement': $scope.showMovement,
-                            'data': $scope.datasetToSendMap,
-                            'from_country': from_country,
-                            'to_country': to_country
-                        });
-
+                    	if (!$routeParams.storyId>0) {
```
Comment (cbotsikas) [inline app/modules/visualization/controllers/visualizationControllers.js]: please do not use $routeParams. Why do you need to check if the chart is (not) in a story? What will happen when used in an other URL where there isn't storyId parameter?
In general, you should not care where the chart is included. Consider adding options in the directive instead.

File: app/modules/visualization/controllers/visualizationControllers.js [hunk 2]
```diff
@@ -893,14 +895,15 @@ angular.module('pcApp.visualization.controllers.visualization', [
...
 
-
-                            if (($scope.mode == 'create') || ($scope.mode == 'edit')) {
-                                if ($scope.visualization.id) {
-                                    document.getElementById("container_graph_" + $scope.visualization.id).innerHTML = "";
-                                } else {
-                                    document.getElementById("container_graph_").innerHTML = "";
-                                }
-                            }
+							if (!$routeParams.storyId>0) {
```
Comment (cbotsikas) [inline app/modules/visualization/controllers/visualizationControllers.js]: please do not use $routeParams.

File: app/modules/visualization/controllers/visualizationControllers.js [hunk 3]
```diff
@@ -910,40 +913,42 @@ angular.module('pcApp.visualization.controllers.visualization', [
...
 
-
-                                if (($scope.mode == 'create') || ($scope.mode == 'edit')) {
-                                    if (document.getElementById("container_graph_" + $scope.visualization.id) != null) {
-                                        document.getElementById("container_graph_" + $scope.visualization.id).innerHTML = document.getElementById("container_graph_" + $scope.visualization.id).innerHTML + "<div class='pie_" + $scope.visualization.id + "' id='pie_" + $scope.visualization.id + "_" + i + "' " + $style + "></div>"
-                                    } else {
-                                        document.getElementById("container_graph_").innerHTML = document.getElementById("container_graph_").innerHTML + "<div class='pie_' id='pie__" + i + "' " + $style + "></div>"
-                                    }
-                                }
+								if (!$routeParams.storyId>0) {
```
Comment (cbotsikas) [inline app/modules/visualization/controllers/visualizationControllers.js]: please do not use $routeParams.

File: app/modules/visualization/controllers/visualizationControllers.js [hunk 4]
```diff
@@ -910,40 +913,42 @@ angular.module('pcApp.visualization.controllers.visualization', [
...
-                                        'radius': radius,
-                                        'innerRadious': innerRadious,
-                                        'font_size': font_size,
-                                        'showLegend': $scope.showLegend, //'showLines': $scope.showLines,
-                                        'showLabels': $scope.showLabels, //'showGrid': $scope.showGrid
-									});
-
-                                    pieObj.render(datasetToSend);
+								
+								if (!$routeParams.storyId>0) {
```
Comment (cbotsikas) [inline app/modules/visualization/controllers/visualizationControllers.js]: please do not use $routeParams.

File: app/modules/visualization/controllers/visualizationControllers.js [hunk 5]
```diff
@@ -3030,58 +3035,60 @@ angular.module('pcApp.visualization.controllers.visualization', [
...
-                                                            barLine.render($scope.numbers1, $scope.eventsToPlot, $scope.mode);
-                                                        }
-
-                                                    });
-                                                } else {
-
-                                                    barLine.render($scope.numbers1, $scope.eventsToPlot, $scope.mode);
-                                                }
-                                            }
+										if (!$routeParams.storyId>0) {
```
Comment (cbotsikas) [inline app/modules/visualization/controllers/visualizationControllers.js]: please do not use $routeParams.

File: app/modules/visualization/controllers/visualizationControllers.js [hunk 6]
```diff
@@ -3173,11 +3180,13 @@ angular.module('pcApp.visualization.controllers.visualization', [
 
                                         $scope.selection = {Keys: $arrayTmp};
                                     }
-
-                                    $scope.plotPieChart();
-                                    
+									
+									if (!$routeParams.storyId>0) {
```
Comment (cbotsikas) [inline app/modules/visualization/controllers/visualizationControllers.js]: please do not use $routeParams.

File: app/modules/visualization/controllers/visualizationControllers.js [hunk 7]
```diff
@@ -3251,66 +3260,68 @@ angular.module('pcApp.visualization.controllers.visualization', [
...
-                                        'showAsPercentatge': $scope.showAsPercentatge,
-                                        'showAstoplines': $scope.showAstoplines,
-                                        'groupby':$scope.groupedby.value,
-                                    });
-
-                                    if (datasetToSend.length>0) {
-	                                    if ($scope.eventsToPlot.length==0) {
-	                                    	barObj.render(datasetToSend, $scope.eventsToPlot);	
+								
+								if (!$routeParams.storyId>0) {
```
Comment (cbotsikas) [inline app/modules/visualization/controllers/visualizationControllers.js]: please do not use $routeParams.

File: app/modules/visualization/controllers/visualizationControllers.js (hunk 1 above)
Comment (mmilaprat) [inline app/modules/visualization/controllers/visualizationControllers.js]: is to avoid problems in the edit mode of stories.

**Pull request merged**